test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -590,7 +563,8 @@ protected bool CheckOutputIsSuffix(string basePath, string outPath, int skip = 0
             return true;
         }
 
-        public bool CompareNumbersWithTolerance(double expected, double actual, int? iterationOnCollection = null, int digitsOfPrecision = DigitsOfPrecision)
```

Comment:
This function is doing two jobs: 1) Comparison and 2) Calling fail if the comparison fails.  

I agree with your point above that there are cases where we want to compare unequal values and make sure they are unequal. Therefore this function should be split into two different functions. One for only doing comparison, and a second one that does the comparison and calls fails depending on whether the comparison was intended to succeed or fail.